File: backup/Implementation/c++/16_문자열압축.cpp
// 이것이 취업을 위한 코딩테스트다 325p
// https://programmers.co.kr/learn/courses/30/lessons/60057
// 2

#include <string>

using namespace std;

int solution(string s) {
    int answer = 1e9;
    if (s.size() == 1)
        return 1;

    for (int cnt = 1; cnt < s.size() / 2 + 1; cnt++) {
        string result;
        string tmp = s.substr(0, cnt);
        int str_cnt = 1;
        for (int i = cnt; i < s.size(); i += cnt) {
            if (tmp == s.substr(i, cnt))
                str_cnt++;
            else {
                if (str_cnt == 1)
                    result += tmp;
                else
                    result += to_string(str_cnt) + tmp;
                tmp = s.substr(i, cnt);
                str_cnt = 1;
            }
        }
        if (str_cnt == 1)
            result += tmp;
        else
            result += to_string(str_cnt) + tmp;

        if (answer > result.size())
            answer = result.size();
    }
    return answer;
}

File: backup/Implementation/python/16_문자열압축.py
# 이것이 취업을 위한 코딩테스트다 325p
# https://programmers.co.kr/learn/courses/30/lessons/60057
# 2

def solution(s):
    answer = 1e9
    if len(s) == 1:
        return 1
    for cnt in range(1, int(len(s) / 2 + 1)):
        result = ""
        str_cnt = 1
        tmp = s[0 : 0 + cnt]
        for i in range(cnt, len(s), cnt):
            if tmp == s[i : i + cnt]:
                str_cnt += 1
            else:
                if str_cnt == 1:
                    result += tmp
                else:
                    result += str(str_cnt) + tmp
                tmp = s[i : i + cnt]
                str_cnt = 1
        if str_cnt == 1:
            result += tmp
        else:
            result += str(str_cnt) + tmp
        answer = min(answer, len(result))
    return answer

print(solution(input()))